src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -263,7 +264,8 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
 
                     using (FileStream fs = File.Create(codePath))
                     {
-                        success = XmlSerializer.GenerateSerializer(serializableTypes, allMappings, fs);
+                        MethodInfo method = typeof(System.Xml.Serialization.XmlSerializer).GetMethod("GenerateSerializer", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);
+                        success = (bool)method.Invoke(null, new object[] { serializableTypes, allMappings, fs });
                     }
                 }
                 catch (UnauthorizedAccessException)
```

Comment:
Should we catch TargetInvocationException to catch the exception thrown by method invoked through reflection?